### PR TITLE
INI: terrible code to test caching generic service responses

### PR DESCRIPTION
**Pull request by eagerterrier**
A rough stab at getting `lru-cache` to handle service generic data fetches and store for 5 minutes.

### CODE REVIEW

File: src/app/routes/cpsAsset/utils/getAdditionalPageData.js [hunk 1]
```diff
@@ -145,14 +146,27 @@ const fetchUrl = async ({ name, path, attachAgent, ...loggerArgs }) => {
...
 
-    return fetchPageData({
-      path,
-      timeout: SECONDARY_DATA_TIMEOUT,
-      agent,
-      ...loggerArgs,
-    })
-      .then(response => validateResponse(response, name))
-      .catch(noop);
+    if (name.indexOf('ecommendations') !== -1){
```
Comment (amoore108) [inline src/app/routes/cpsAsset/utils/getAdditionalPageData.js]: Small typo here for `recommendations`.

May need to take into account the other data types that could be cached, `mostRead`, `mostWatched`? Or is this just to test `recommendations`?

EDIT: Actually just re-read, this should cache all the data types **apart** from recommendations? Just the typo needs fixed then!

Comment (eagerterrier) [reply src/app/routes/cpsAsset/utils/getAdditionalPageData.js]: ah, yes, I realised that some recommendations were Recommendations and some were recommendations and didn't want extra weight of a case insensitive regex, so I cut off first letter 😂 

tbh, couldn't remember if string comparison was case insensitive by default. Just wanted the thing just to work. I tested this by outputting `cache.keys()` and it was definitely using the cache after first load, which is good.

You are right, needs a bit of refactoring/polishing to help it be production ready

File: src/app/routes/utils/fetchPageData/withCache.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  ...loggerArgs}) => fetchPageData({path,
+  timeout,
+  shouldLogFetchTime,
+  agent,
+  optHeaders,
+  cache,
+  ...loggerArgs});
+
+console.log('caache keys', cache.keys());
+console.log('caache keys fetchPageData', withCache);
```
Comment (amoore108) [inline src/app/routes/utils/fetchPageData/withCache.js]: Some console logs could be removed, but just nit-picking.

**Review by amoore108 — APPROVED** [review]
I think this looks great 👍🏼 Worth trying on Test to see how it performs.

File: scripts/bundleSize/bundleSizeConfig.js [hunk 3]
```diff
@@ -6,5 +6,5 @@ module.exports = {
   // need for frequent changes as bundle sizes fluctuate.
 
   MIN_SIZE: 763,
-  MAX_SIZE: 1202,
+  MAX_SIZE: 1221,
```
Comment (amoore108) [inline scripts/bundleSize/bundleSizeConfig.js:9]: I'd try setting these to `768` and `1209`. Ran a `yarn build` locally on this branch and those should be calculated min/max values after the `lru-cache` update.

Comment (eagerterrier) [reply scripts/bundleSize/bundleSizeConfig.js:9]: ok, I followed the guidance on adding 5, as my figure for burmese was 1216. Were you getting lower figure?

Comment (amoore108) [reply scripts/bundleSize/bundleSizeConfig.js:9]: Yea I got `1204` in my local build 🤷🏼

File: src/app/routes/cpsAsset/utils/getAdditionalPageData.js [hunk 4]
```diff
@@ -145,7 +146,19 @@ const fetchUrl = async ({ name, path, attachAgent, ...loggerArgs }) => {
   try {
     const agent = attachAgent ? await getAgent() : null;
 
-    return fetchPageData({
+    if (name.indexOf('ecommendations') !== -1) {
```
Comment (karinathomasbbc) [inline src/app/routes/cpsAsset/utils/getAdditionalPageData.js]: Would this work?
```suggestion
    if (['recommendations'].includes(name.toLowercase())) {
```

**Review by karinathomasbbc — APPROVED** [review]
Let's try this and see 🤞 !!

File: src/app/routes/cpsAsset/utils/getAdditionalPageData.js (hunk 4 above)
Comment (eagerterrier) [reply src/app/routes/cpsAsset/utils/getAdditionalPageData.js]: that didn't work! It messed things up. No idea why.

Comment (karinathomasbbc) [reply src/app/routes/cpsAsset/utils/getAdditionalPageData.js]: Oh nos! I've checked out the branch - lemme see if I can figure it out

**Pull request merged**